src/plugins/borrow-plugins/plugins/aave/BorrowEngineFactory.js

```diff
@@ -122,22 +144,79 @@ export const makeBorrowEngineFactory = (blueprint: BorrowEngineBlueprint) => {
       },
 
       async deposit(request: DepositRequest): Promise<ApprovableAction> {
-        return {
-          networkFee: {
-            currencyCode: wallet.currencyInfo.currencyCode,
-            nativeAmount: '2100000000000000'
-          },
-          approve: async () => {}
+        const { nativeAmount, tokenId, fromWallet } = asDepositRequest(request)
+
+        const token = asEdgeToken(tokenId)
+        const tokenAddress = asTokenAddress(token)
+
+        const asset = tokenAddress
+        const amount = BigNumber.from(nativeAmount)
+        const onBehalfOf = (await fromWallet.getReceiveAddress()).publicAddress
+
+        const gasPrice = await aaveNetwork.provider.getGasPrice()
+
+        const tokenContract = await aaveNetwork.makeTokenContract(tokenAddress)
+
+        const txCallInfos: CallInfo[] = []
+
+        const allowance = await tokenContract.allowance(onBehalfOf, aaveNetwork.lendingPool.address)
+        if (!allowance.sub(nativeAmount).gte(0)) {
+          const approveTx = await tokenContract.populateTransaction.approve(aaveNetwork.lendingPool.address, ethers.constants.MaxUint256, {
+            gasLimit: '500000',
+            gasPrice
+          })
+          txCallInfos.push({
+            tx: approveTx,
+            wallet,
+            token,
+            metadata: {
+              name: 'AAVE',
+              category: 'expense:approval',
```

Comment:
Are these existing categories in the app? I think Edge has `expense:fee`, but not `expense:approval`. I would be very confused if I got `expense:approval` on my tax spreadsheets.

Reply:
We do have `expense:Network Fee` being generated in other parts of the app, so let's use that one.

Reply:
Should we just exclude the sub-category?

Reply:
We should ask @paullinator what he wants, since the categories are important for a good UX. I personally vote for `expense:Network Fee`, which we are already using.